synapse/storage/database.py

```diff
@@ -179,7 +179,7 @@ def __getattr__(self, name):
 
 
 # The type of entry which goes on our after_callbacks and exception_callbacks lists.
-_CallbackListEntry = Tuple[Callable[..., None], Iterable[Any], Dict[str, Any]]
+_CallbackListEntry = Tuple[Callable[..., object], Iterable[Any], Dict[str, Any]]
```

Comment:
Curious why this is `object` and not `Any`?

Reply:
It's to make mypy complain if we ever try to use the return values of the callbacks.

Everything is assignable to `object`, but not the other way around, whereas `Any` is assignable to everything without any checks.

Reply:
`object` is the supertype of everything is what I think you're saying, so we'd get shouted at if we tried to do `x: int = (something returning object)` whereas `Any` is entirely untyped & unchecked.

I think this makes sense.

Reply:
Does this mean we should generally prefer `object` over `Any`?

Reply:
I haven't a clue. I imagine it's on a case by case basis. `Any` is certainly a last resort for me, since it's equivalent to not annotating types at all.